src/utils/datamap.js

```diff
@@ -0,0 +1,20 @@
+import _ from "lodash";
+import iota from "services/iota";
+import Encryption from "utils/encryption";
+
+const generate = (size, handle) => {
+  const keys = _.range(1, size);
+  const handleInTrytes = iota.utils.toTrytes(handle);
+  return _.reduce(
+    keys,
+    (hash, n) => {
+      const encryptedHash = Encryption.encrypt(hash[n - 1], handle);
+      const encryptedHashInTrytes = iota.utils.toTrytes(encryptedHash);
+      hash[n] = encryptedHashInTrytes.substr(0, 81);
+      return hash;
+    },
+    { 0: handleInTrytes.substr(0, 81) }
+  );
+};
```

Comment:
In my next pull I will optimize this by storing the datamap in storage (https://github.com/marcuswestin/store.js/) and then somehow polling all the IOTA transactions in parts